lib/simulator-xcode-9.js

```diff
@@ -160,6 +149,37 @@ class SimulatorXcode9 extends SimulatorXcode8 {
     }
   }
 
+  /***
+   * Boots simulator and opens simulators UI Client if not already opened.
+   *
+   * @param {?String} isUiClientRunning - process id of simulator UI client.
+   * @param {Object} opts - arguments to start simulator UI client with.
+   */
+  async launchWindow (isUiClientRunning, opts) {
+    await this.bootSimulator();
+    if (!isUiClientRunning) {
+      await this.startUIClient(opts);
+    }
+  }
+
+  /**
+   * Boots simulator if not already booted.
+   */
+  async bootSimulator () {
+    const {state: serverState} = await this.stat();
```

Comment:
I would rather catch the exception from `simctl.bootDevice` and check stderr to decide whether to retry or not. Calling `stat` adds a performance hit.

Reply:
@mykola-mokhnach are you looking for something like 
`async bootSimulator () {
    try {
      log.info(`Booting Simulator with UDID '${this.udid}'...`);
      await this.simctl.bootDevice();
    } catch (err) {
      if (!(err.stdout || '').includes('Unable to boot device in current state: Booted') &&
          !(err.stderr || '').includes('Unable to boot device in current state: Booted')) {
        await retryInterval(3, 2000, async () => await this.simctl.bootDevice());
      } else {
        log.warn(`Error while booting simulator: ${err.stdout || err.stderr}.`);
      }
    }
  }`
?